app/routes/routes.py

```diff
@@ -0,0 +1,63 @@
+from flask import request, jsonify
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+
+from . import tasks
+from . import goals
+
+
+def bp_model(bp_name):
+    model = {"tasks": Task, "goals": Goal}.get(bp_name, None)
+    return model, bp_name[:-1]
+
+
+@tasks.route("", methods=["GET"])
+@goals.route("", methods=["GET"])
+def get_all_items():
+    model, _ = bp_model(request.blueprint)
+
+    by = request.args.get("by")
+    by = getattr(model, by) if by in ("title", "task_id", "goal_id") else model.title
+    sort = request.args.get("sort")
+    order_by = getattr(by, sort)() if sort in ("asc", "desc") else None
```

Comment:
While I do generally like this pattern, I do think that the way that function invocation `()` needs to be supplied on line 23 feels awkward enough that I've been calling it out as a possible readability issue. A lot of this comes down to that `()` being a result of the underlying SQLAlchemy implementation, making this code tightly coupled with the SQLAlchemy code, which is considered a bit of a "code smell".

The issue would mainly come into play if SQLAlchemy changes their class structure, but mostly it's specific enough that I would not assume a reader of your code would _know_ why this line uses `()` while the line before it does not. But a simple comment explaining that would be sufficient.

There's a tough balance of what you can assume a reader of your code "knows", and if something takes a reader some time to puzzle out (as I did when I first saw this pattern in another student's submission), then other people will likely need to take time to figure it out. Which makes the code a bit harder to maintain.

But note, I didn't recommend documenting other stuff I didn't know about because those in my mind are just Flask/SQLAlchemy features I reasonably _should_ have known about, while this feels more like a SQLAlchemy _implementation detail_, and thus not as reasonable to expect a reader to know. It's very handwavy.

Reply:
this is one of the more awkward bits of my design, and i would like to revise it. i had some other solutions previously that were more obvious, i might return to those.